### DISCUSSION TITLE
auto-cpufreq + undervolt = true?

### DISCUSSION THREAD

**Discussion by kristian290598** [AdnanHodzic/auto-cpufreq#375] [Q&A]
Hello, I already have auto-cpufreq installed. I have a laptop that are used for school so battery life is king of course, will an undervolt result in longer battery life? Does it conflict with the package? Is there a program like PCmark for linux to test battery life?

Thank you for the awesome work with this package

**Comment by AdnanHodzic** [comment]
Thanks, I don't know, as I never used undervolt. auto-cpufreq by itself should greatly extend your better battery life, I suggest you play with different [configuration options ](https://github.com/AdnanHodzic/auto-cpufreq#configuring-auto-cpufreq)to extend it even more. As using other tools can only potentially conflict with it and lead to negative results.

**Comment by pavelskipenes** [comment]
Power consumption is linearly dependent on the frequency and squared dependent (if that's correct way of phrasing it) on voltage. In other words cutting voltage in 2 results in 4 times lower power consumption while cutting frequency in 2 will cut the power consumption in 2.
```
power_consumption = some_random_constant * frequency * voltage * voltage;
```
